### ISSUE TITLE
Investigate effect of panic=abort on build time

### ISSUE THREAD

**Issue by brson** [tikv/tikv#4173]
It looks reasonable to me that tikv can potentially abort on panic in release mode. Investigate how it impacts compile time and whether it is a reasonable thing for the product to do.

**Comment by brson** [comment]
I did an experiment comparing the compile time and tikv-server binary size. The command I used was `cargo build --all --release`.

`panic = "abort"` resulted in a 3.3% compile time improvement and a 1.6% binary size improvement.

Seems like a worthy thing to try to put into the "dev" profile and a ["dev+optimized"](https://github.com/tikv/tikv/issues/4189) profile. Still need to investigate whether the test suite depends on unwinding, figure out what to do about that.

Raw results below.

before:

```
real    25m13.237s
user    58m28.394s
sys     2m1.420s
```

tikv-server size: 311513464 bytes

after:

```
real    24m23.134s
user    57m37.594s
sys     2m2.737s
```

tikv-server size: 306505344 bytes
